src/bin/stratisd.rs

```diff
@@ -46,6 +49,25 @@ fn write_or_panic(err: StratisError) -> () {
     }
 }
 
+/// Given a udev event check to see if it's an add and if it is return the device node and
+/// device number.
+fn handle_udev_add(event: &libudev::Event) -> Option<(PathBuf, u64)> {
+    match event.event_type() {
+        libudev::EventType::Add => {
+            info!("Hot plug add {:?}{:?}",
+                  event.device().devnum(),
+                  event.device().devnode());
+            if let Some(devnode) = event.device().devnode() {
```

Comment:
Probably a true Rustacean would go with:

```
let device = event.device();
device.devnode().and_then(|devnode| device.devnum().and_then(|devnum| (PathBuf::from(devnode), devnum)))
```
or something similar as whole body of function.